backend/crypt/crypt.go

```diff
@@ -351,6 +367,9 @@ func (f *Fs) put(ctx context.Context, in io.Reader, src fs.ObjectInfo, options [
 	if err != nil {
 		return nil, err
 	}
+	if f.opt.NoDataEncryption {
```

Comment:
This is doing a lot more work than it needs to.

I think if you put this at the start of the function then it should work

```go
if f.opt.NoDataEncryption {
    return put(ctx, wrappedIn, f.newObjectInfo(src, nonce{}), options...)
}
```